File: CLI/HelpCommand.cs
using System;
using System.Threading.Tasks;

namespace CLI
{
    internal class HelpCommand : BaseCommand
    {
        public const string COMMAND_NAME = "help";
        private readonly BaseCommand[] _commands;

        public HelpCommand(BaseCommand[] commands) : base(COMMAND_NAME)
        {
            _commands = commands;
        }

        public override Task Run(string[] args)
        {
            PrintHelp();

            return Task.FromResult(true);
        }

        public override void PrintHelp()
        {
            Console.WriteLine(@"CSGO Demo Manager CLI");
            Console.WriteLine(@"");
            Console.WriteLine($@"    Usage: {Program.ExeName} <command> [arguments]");
            Console.WriteLine(@"");
            Console.WriteLine(@"The commands are:");
            Console.WriteLine(@"");
            foreach (BaseCommand command in _commands)
            {
                Console.WriteLine($@"    {command.GetName(),-20} {command.GetDescription()}");
            }
        }
    }
}

File: src/cli/commands/help-command.ts
import { Command } from './command';
import { commands } from '../commands';
import pkg from '../../../package.json';

export class HelpCommand extends Command {
  public static Name = 'help';

  public getDescription() {
    return 'Show CLI help usage';
  }

  public printHelp() {
    console.log(`CS Demo Manager ${pkg.version}`);
    console.log('');
    console.log('\tUsage: csdm <command> [arguments]');
    console.log('');
    console.log('The commands are:');
    console.log('');
    for (const [commandName, CommandToRun] of Object.entries(commands)) {
      const command = new CommandToRun([]);
      console.log(`\t${commandName.padEnd(10)} \t${command.getDescription()}`);
    }
  }

  public async run() {
    this.printHelp();

    return Promise.resolve();
  }
}
